gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/schema/SafIdtScheme.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.gateway.security.service.schema;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.zuul.context.RequestContext;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import org.zowe.apiml.auth.Authentication;
+import org.zowe.apiml.auth.AuthenticationScheme;
+import org.zowe.apiml.gateway.security.service.AuthenticationService;
+import org.zowe.apiml.gateway.security.service.SafAuthenticationService;
+import org.zowe.apiml.security.common.token.QueryResponse;
+import org.zowe.apiml.security.common.token.TokenAuthentication;
+
+import java.util.Date;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+/**
+ * The scheme allowing for the safIdt authentication scheme.
+ */
+@Component
+@RequiredArgsConstructor
+public class SafIdtScheme implements AbstractAuthenticationScheme {
+    private final AuthenticationService authenticationService;
+    private final SafAuthenticationService safAuthenticationService;
+
+    @Override
+    public AuthenticationScheme getScheme() {
+        return AuthenticationScheme.SAF_IDT;
+    }
+
+    @Override
+    public AuthenticationCommand createCommand(Authentication authentication, Supplier<QueryResponse> tokenSupplier) {
+        // Same behavior as for the ZosmfScheme.
+        final QueryResponse queryResponse = tokenSupplier.get();
+        final Date expiration = queryResponse == null ? null : queryResponse.getExpiration();
+        final Long expirationTime = expiration == null ? null : expiration.getTime();
+        return new SafIdtCommand(expirationTime);
+    }
+
+    @RequiredArgsConstructor
+    public class SafIdtCommand extends AuthenticationCommand {
+        private final Long expireAt;
+
+        @Override
+        public void apply(InstanceInfo instanceInfo) {
+            final RequestContext context = RequestContext.getCurrentContext();
+
+            Optional<String> jwtToken = authenticationService.getJwtTokenFromRequest(context.getRequest());
+            jwtToken.ifPresent(token -> {
+                TokenAuthentication authentication = authenticationService.validateJwtToken(jwtToken.get());
+                if (authentication.isAuthenticated()) {
+                    String safIdt = safAuthenticationService.generateSafIdt(token);
+
+                    // remove authentication part
+                    context.addZuulRequestHeader("X-SAF-Token", safIdt);
+                }
```

Comment:
The other commands work in a way that they exchange one form of authentication for another. This one adds second form to the first one. Is that intended?